spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala

```diff
@@ -261,10 +270,15 @@ trait CommonJobExecution extends ProjectMetadata {
     }
   }
 
-  private def getReportVersion[T](jobConfig: JobConfigParser[T], dataset: Dataset)(implicit fsUtils: FileSystemVersionUtils): Int = {
+  private def getReportVersion[T](jobConfig: JobConfigParser[T], dataset: Dataset)(implicit hadoopConf: Configuration): Int = {
     jobConfig.reportVersion match {
       case Some(version) => version
       case None =>
+
+        // publishFs for this specific feature (needed for missing reportVersion until reusable
```

Comment:
I am not even sure what the comment means. 🤔 @dk1844 ?

Reply:
I believe the comment is needed. Maybe it should be reworded though if it is unclear. The gist of the meaning is this:

Everywhere else, one would want to reuse existing "Publish FS" that will reside in `pathCfg.publish.fileSystem` in this situation (to get an instance of `HadoopFsUtils` bound to this FS). However, here, you cannot do it, because `pathCfg` object is not yet available at this point. So, you the would-be-publishFs is here obtained temporarily from the `dataset.hdfsPublishPath.`

That's all. The comment just meant to explain why seemingly the call of `dataset.hdfsPublishPath` is pointlessly repeated here and also later in the factory method of PathConfig. It just could not be reused reasonably.

Reply:
My suggestion for the comment here is:
```scala
// Since "pathConfig.publish.fileSystem" is not available at this point yet, the a temporary publish FS is used here instead
```